deepmd/pt/utils/lmdb_dataset.py

```
@@ -30,36 +31,133 @@
+            result[key] = [d[key] for d in batch]
+        elif key == "type":
+            continue
+        elif batch[0][key] is None:
+            result[key] = None
+        else:
+            tensors = [torch.as_tensor(d[key]) for d in batch]
+            with torch.device("cpu"):
+                if key in _ATOMWISE_MIXED_BATCH_KEYS:
+                    result[key] = torch.cat(tensors, dim=0)
+                else:
+                    result[key] = collate_tensor_fn(tensors)
```

Ruff (0.15.12)

[error] Undefined name `collate_tensor_fn`

(F821)